hip-service/Link/Patient/PatientVerification.cs

```diff
@@ -1,27 +1,84 @@
+        public PatientVerification(IConfiguration configuration)
         {
-            this.otpVerification = otpVerification;
+            this.configuration = configuration;
         }
-
-        public async Task<Error?> SendTokenFor(Session session)
+        
+        public async Task<OtpMessage> SendTokenFor(Session session)
         {
-            return await otpVerification.GenerateOtp(session);
+            var uri = new Uri(configuration.GetConnectionString("OTPGenerationConnection"));
+            var verifyMessage = await PostCallToOTPServer(uri
+                , CreateHttpContent(session)).ConfigureAwait(false);
+            return verifyMessage.ValueOr((OtpMessage) null);
         }
         
-        public async Task<Error?> Verify(string sessionId, string value)
+        public async Task<OtpMessage> Verify(string sessionId, string value)
         {
-            return await otpVerification.CheckOtpValue(sessionId, value);
+            var uri = new Uri(configuration.GetConnectionString("OTPVerificationConnection"));
+            var verifyOtpRequest = new OtpVerificationRequest(sessionId, value);
+            var verifyMessage = await PostCallToOTPServer(uri
+                , CreateHttpContent(verifyOtpRequest)).ConfigureAwait(false);
+            return verifyMessage.ValueOr((OtpMessage) null);
+        }
+        
+        private static HttpContent CreateHttpContent<T>(T content)  
+        {
+            var json = JsonConvert.SerializeObject(content);  
+            return new StringContent(json, Encoding.UTF8, "application/json");  
+        }
+        
+        private async Task<Option<OtpMessage>> PostCallToOTPServer (Uri serverUrl, HttpContent content)
+        {
+            var clientHandler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (sender, cert, chain, sslPolicyErrors) => true
+            };
+            var httpClient = new HttpClient(clientHandler);
+            httpClient.DefaultRequestHeaders.Accept.Add(
+                new MediaTypeWithQualityHeaderValue("application/json"));
+            try
+            {
+                var response = await httpClient.PostAsync(serverUrl.ToString()
+                    , content);
+            
+                if (response.IsSuccessStatusCode)
+                {
+                    return Option.None<OtpMessage>();
+                }
+
+                if (!response.IsSuccessStatusCode)
+                {
+                    var responseContent = response.Content;
+                    using var reader = new StreamReader(await responseContent.ReadAsStreamAsync());
+                    var result = await reader.ReadToEndAsync().ConfigureAwait(false);
+                    var otpMessage = JsonConvert.DeserializeObject<OtpMessage>(result);
+                    return Option.Some(otpMessage);
+                }
+                return Option.Some(new OtpMessage(ErrorCode.ServerInternalError.ToString()
```

Comment:
Why can't just return Option<Error> from this method, and decide what to return at the controller?

Reply:
But we need the message why it failed